bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Bookie.java

```diff
@@ -357,10 +367,15 @@ private void checkEnvironment(ZooKeeper zk) throws BookieException, IOException
                 // 1) new environment or
                 // 2) done only metadata format and started bookie server.
             }
-            checkDirectoryStructure(journalDirectory);
 
-            if(!newEnv){
-                journalCookie.verify(masterCookie);
+            for (File journalDirectory : journalDirectories) {
+                checkDirectoryStructure(journalDirectory);
+            }
```

Comment:
I don't remember exactly, though, I was just replicating the same steps being done on the single journal directory